src/preflight/handler.js

```diff
@@ -385,6 +320,71 @@ export const preflightAudit = async (context) => {
       await saveIntermediateResults(result, 'DOM-based audit');
     }
 
+    if (!checks || checks.includes(AUDIT_LINKS)) {
+      // Internal link checks
+      const internalLinksStartTime = Date.now();
+      const internalLinksStartTimestamp = new Date().toISOString();
+      // Create links audit entries for all pages
+      normalizedUrls.forEach((url) => {
+        const pageResult = resultMap.get(url);
+        pageResult.audits.push({ name: AUDIT_LINKS, type: 'seo', opportunities: [] });
+      });
+
+      const { auditResult } = await runInternalLinkChecks(scrapedObjects, context, {
+        pageAuthToken: `token ${pageAuthToken}`,
+      });
+      if (isNonEmptyArray(auditResult.brokenInternalLinks)) {
+        auditResult.brokenInternalLinks.forEach(({ pageUrl, href, status }) => {
+          const audit = resultMap.get(pageUrl).audits.find((a) => a.name === AUDIT_LINKS);
+          audit.opportunities.push({
+            check: 'broken-internal-links',
+            issue: {
+              url: href,
+              issue: `Status ${status}`,
+              seoImpact: 'High',
+              seoRecommendation: 'Fix or remove broken links to improve user experience and SEO',
+            },
+          });
+        });
+      }
+      const internalLinksEndTime = Date.now();
+      const internalLinksEndTimestamp = new Date().toISOString();
+      const internalLinksElapsed = ((internalLinksEndTime - internalLinksStartTime) / 1000)
+        .toFixed(2);
+      log.info(`[preflight-audit] site: ${site.getId()}, job: ${jobId}, step: ${normalizedStep}. Internal links audit completed in ${internalLinksElapsed} seconds`);
+
+      timeExecutionBreakdown.push({
+        name: 'links',
+        duration: `${internalLinksElapsed} seconds`,
+        startTime: internalLinksStartTimestamp,
+        endTime: internalLinksEndTimestamp,
+      });
+
+      await saveIntermediateResults(result, 'internal links audit');
+    }
+
+    if (!checks || checks.includes(AUDIT_LINKS)) {
```

Comment:
isn't this condition same as here https://github.com/adobe/spacecat-audit-worker/pull/978/files#diff-56d6bf732d419d5caa8e25f33ec139ba9c1d6ab7e1dcdab550db87e0adfe9203R323 ?

Reply:
@anagarwa , you are right - grouped them together - https://github.com/adobe/spacecat-audit-worker/pull/978/commits/eb8cb73f604fec2c3ab34da901aa7c9bd28dde4e